warmup/summary_statistics.py

```diff
@@ -57,21 +65,54 @@ def collect_summary_statistics(data_dictionaries, delta, steady_state):
                 # steady state. However, the last segment may be equivalent to
                 # its adjacent segments, so we first need to know which segments
                 # are steady-state segments.
+                if data_dictionaries[machine]['classifications'][key][p_exec] == 'no steady state':
+                    continue
+                # Capture the last steady state segment for bootstrapping.
```

Comment:
Similar (but not identical) code appears in a number of places, for example, where we choose whether to plot data in grey or black. If we put an XXX in here I would worry that we might miss cases elsewhere. We probably should try to factor it out somehow, but since all the "similar" cases are subtly different, this will need some careful thought.